services/web/client/source/resource/qxapp/project-v1.0.0.json

```diff
@@ -0,0 +1,211 @@
+              ]
+            },
+            "version": {
+              "type": "string",
+              "description": "semantic version number of the node",
+              "pattern": "^(0|[1-9]\\d*)(\\.(0|[1-9]\\d*)){2}(-(0|[1-9]\\d*|\\d*[-a-zA-Z][-\\da-zA-Z]*)(\\.(0|[1-9]\\d*|\\d*[-a-zA-Z][-\\da-zA-Z]*))*)?(\\+[-\\da-zA-Z]+(\\.[-\\da-zA-Z-]+)*)?$",
+              "examples": [
+                "1.0.0",
+                "0.0.1"
+              ]
+            },
+            "label": {
+              "type": "string",
+              "description": "human readable name for the node to be displayed instead of the default name"
+            },
+            "inputs": {
+              "type": "object",
+              "description": "An opaque data structure describing the input configuration as well as the internal state of the node."
+            },
+            "inputNodes": {
+              "type": "array",
+              "description": "list of UUIDS of input nodes providing input data to this node",
+              "items": {
+                "type":"string",
+                "pattern": "^[-_a-zA-Z0-9]+$",
+                "description": "node UUID"
+              }
+            },
+            "outputs": {
+              "type": "object",
+              "description": "nodes providing static (non api output) store their output here after computation is complete.",
+              "patternProperties": {
+                "^[-_a-zA-Z0-9]+$": {
+                  "oneOf": [
+                    {
+                      "type": [
+                        "integer",
+                        "boolean",
+                        "string",
+                        "number"
+                      ]
+                    },
+                    {
+                      "type": "object",
+                      "additionalProperties": false,
+                      "required": [
+                        "store",
+                        "path"
+                      ],
+                      "properties": {
+                        "store": {
+                          "type": "string"
+                        },
+                        "path": {
+                          "type": "string"
+                        }
+                      }
+                    }
+                  ]
+                }
+              }
+            },
+            "position": {
+              "type": "object",
+              "additionalProperties": false,
+              "required": [
+                "x",
+                "y"
+              ],
+              "properties": {
+                "x": {
+                  "type": "integer"
+                },
+                "y": {
+                  "type": "integer"
+                }
+              }
+            },
+            "parent": {
+              "type": "string",
+              "pattern": "^[-_a-zA-Z0-9]+$",
+              "description": "pointing to a container node"
+            }
```

Comment:
What is this parent for?

Reply:
this would be pointing to the UUID of a container node, representing the hierarchy of the project.